app/main.py

```
@@ -1,25 +1,140 @@
-from abc import ABC
+from __future__ import annotations
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
+            name: str
+    ) -> None:
+        self.private_name = "_" + name
+        self.public_name = name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: Slide
+    ) -> int:
+        return getattr(instance, self.private_name)
+
+    def __set__(
+            self,
+            instance: SlideLimitationValidator,
+            value: int
+    ) -> None:
+        if instance.validate(value, self.min_amount, self.max_amount):
+            setattr(instance, self.private_name, value)
+        else:
+            setattr(instance, self.private_name, False)
```

Instead of delegating the logic of the validation to the `instance`, perform all required checks in the `__set__` method itself. You could raise an error and process it in the `can_access` method of the `Slide` class.